experimenter/database_communication.py

```diff
@@ -196,18 +197,15 @@ def add_to_pipeline_runs_mongo(self, pipeline_run: dict, collection_name: str):
         else:
             logger.info("\n\nWARNING: PIPELINE_RUN ALREADY EXISTS IN DATABASE. NOTHING WRITTEN.\n\n")
 
-    def add_to_pipelines_mongo(self, new_pipeline: Pipeline) -> bool:
+    def add_to_pipelines_mongo(self, new_pipeline: EZPipeline) -> bool:
         """
         Function to add a pipeline to the mongodb database of pipelines.
         :param new_pipeline: the new pipeline to add to mongo
         :return False if the database already contains it, True if the pipeline was added to the database
         """
         db = self.mongo_client.metalearning
         collection = db.pipelines
-        if type(new_pipeline) != dict:
-            new_pipeline_json = new_pipeline.to_json_structure()
-        else:
-            new_pipeline_json = new_pipeline
```

Comment:
I have verified that all places where `add_to_pipelines_mongo` is called are passing an `EZPipeline` object to it, so this check is no longer needed.